services/etcd.go

```diff
@@ -318,6 +318,46 @@ func RunEtcdSnapshotSave(ctx context.Context, etcdHost *hosts.Host, prsMap map[s
 	return nil
 }
 
+func DownloadEtcdSnapshot(ctx context.Context, etcdHost *hosts.Host, prsMap map[string]v3.PrivateRegistry, etcdSnapshotImage string, name string, once bool, es v3.ETCDService) error {
+	log.Infof(ctx, "[etcd] Get snapshot [%s] on host [%s]", name, etcdHost.Address)
+	s3Backend := es.BackupTarget.S3BackupTarget
+	if len(s3Backend.Endpoint) == 0 || len(s3Backend.BucketName) == 0 {
+		return fmt.Errorf("failed to get snapshot [%s] from s3 on host [%s], invalid s3 configurations", name, etcdHost.Address)
+	}
+	imageCfg := &container.Config{
+		Cmd: []string{
+			"/opt/rke-tools/rke-etcd-backup",
+			"etcd-backup",
+			"download",
+			"--name", name,
+			"--endpoints=" + etcdHost.InternalAddress + ":2379",
```

Comment:
Why do you need the endpoints for download ?

Reply:
removed.